ott/core/sinkhorn_lr.py

```diff
@@ -114,20 +118,11 @@ def solution_error(
   return err
 
 
-def compute_criterion(
-    q: jnp.ndarray, r: jnp.ndarray, g: jnp.ndarray, q_prev: jnp.ndarray,
-    r_prev: jnp.ndarray, g_prev: jnp.ndarray, gamma: float
-):
-  err_1 = ((1 / gamma) ** 2) * (kl(q, q_prev) + kl(q_prev, q))
-  err_2 = ((1 / gamma) ** 2) * (kl(r, r_prev) + kl(r_prev, r))
-  err_3 = ((1 / gamma) ** 2) * (kl(g, g_prev) + kl(g_prev, g))
-  criterion = err_1 + err_2 + err_3
-  return criterion
-
-
-def kl(q1, q2):
-  ratio = jnp.log(q1) - jnp.log(q2)
-  return jnp.sum(q1 * ratio)
+def kl(q1, q2, clipping_value=1e-8):
```

Comment:
I like this implementation better than the one I in `quad_problems.py`, would consider re-using it there as well.

Reply:
Also would add typehints for this.